impl/utility.cxx

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifdef _WIN32
+#include <windows.h>
+#include <dbghelp.h>
+#endif
+#ifndef _MSC_VER
+#include <cxxabi.h>
+#endif
+
+#include <string>
+#include <system_error>
+#include <vector>
+
+#include "substrate/utility"
+
+namespace substrate
+{
+	std::string decode_typename(const char *const mangled_name)
+	{
+#ifndef _MSC_VER
+		int status{};
+		auto * const demangler{ abi::__cxa_demangle(mangled_name, nullptr, nullptr, &status) };
+		if (status == -1)
+			throw std::runtime_error("A memory allocation failure occurred");
+		else if (status == -3)
+			return { mangled_name };
```

Comment:
```suggestion
return {manged_name};
```

Reply:
I'm not sure what this suggestion means?